Automattic/wp-calypso
# Domains: Fix all navigation links inside domains

Some domain links/navigation are not working properly in some specific contexts - for example, a few "All domains" navigation routes. For them to work properly, we need to pass the `currentRoute` argument, so that the navigation can go back to the correct page and keep the components in the same context. This PR fixes this behavior.

## Testing Instructions
- Go to "All domains";
- Select any domain and navigate through a few management routes. Some examples:
  - DNS records > Manage > Add new DNS > Cancel;
  - Transfer > Click on the "All domains" breadcrumb item;
  - Contact Information > Edit > Click on the "All domains" breadcrumb item;
- Ensure these are all working;
- Ensure all unit tests pass.

### Before
https://github.com/Automattic/wp-calypso/assets/18705930/cbd795cf-d97e-4fd4-80b3-87e652621a72

### After
https://github.com/Automattic/wp-calypso/assets/18705930/364f98e6-0fd6-4c48-b4c6-fd4d4264b205

## Pre-merge Checklist
